pilot/pkg/model/config.go

```diff
@@ -547,7 +547,6 @@ func ResolveShortnameToFQDN(host string, meta ConfigMeta) Hostname {
 // MostSpecificHostMatch compares the elements of the stack to the needle, and returns the longest stack element
 // matching the needle, or false if no element in the stack matches the needle.
 func MostSpecificHostMatch(needle Hostname, stack []Hostname) (Hostname, bool) {
-	sort.Sort(Hostnames(stack))
```

Comment:
If you're doing the most specific match you still need sorting. As-is, this is just "give me the first match".

Reply:
Yep I know. I moved the sort to the initialization code, as there is no point in sorting the same array for every call.